### ISSUE TITLE
[Bug]: Removing value in numeric cell results in empty string

### ISSUE THREAD

**Issue by Dissident26** [handsontable/handsontable#10738]
### Describe the bug

Removing value from cell with editor opened results in empty string is set to the cell value.

Repro steps: double click on a cell with type "numeric" and see console.log, value should not be a string in "numeric" type cell.

### Video/Screenshots

![image](https://github.com/handsontable/handsontable/assets/60570678/4215c92b-dec9-4974-92b0-36407fdc29d2)

### Provide a link to the demo with the bug reproduction

https://jsfiddle.net/r8n3q205/8/

### Handsontable version

"^13.1.0"

### Your environment

chrome, windows

**Comment by adrianszymanski89** [comment]
Hi @Dissident26 

I wouldn't consider it as a bug. All our built-in validators treat `null` or empty strings as valid. Only the values not in the correct format for the defined type are treated as invalid.

**Comment by Dissident26** [comment]
> Hi @Dissident26
> 
> I wouldn't consider it as a bug. All our built-in validators treat `null` or empty strings as valid. Only the values not in the correct format for the defined type are treated as invalid.

Hi @adrianszymanski89, thanks for answering. 

Yes, I do understand that, and it's fine to check on falsy values. But in my opinion it's the matter of type, not the falsy values.
When you have a type validation on `number | null` it's not obvious that you would get a value of type `string` eventually.

**Comment by AMBudnik** [comment]
You are right, @Dissident26. While checking the docs I did not find clear information on what we get.

We say at http://localhost:8080/docs/javascript-data-grid/keyboard-shortcuts/#selection-keyboard-shortcuts that

![Zrzut ekranu 2024-01-19 o 13 49 43](https://github.com/handsontable/handsontable/assets/10757813/b6608ff9-5ce3-4ba1-8745-ea03ec75c255)

and in general, cleaning is rather setting up `null`, as in `clear()` method that we can test here https://jsfiddle.net/3r9sa28L/.

But then when you select a cell it is treated more like an action of a cell editor and there we put an empty string.

So maybe we could add an option to allow developers setting up what the `clear` should mean.

**Comment by adrianspdev** [comment]
Hi @Dissident26,

Thanks for the report.

Two distinct "remove" flows, by design:

1. Delete/Backspace (no editor) or clear() → cell becomes null. Grid-level clear.
2. Open editor → erase → confirm → cell becomes "". TextEditor.getValue() returns this.TEXTAREA.value (always string); NumericEditor inherits it unchanged.

numericValidator coerces null/undefined/"" into same "empty" branch and passes when allowEmpty: true (default), so validation can't normalize type.

No auto-coerce "" → null because: back-compat, divergence across editors extending TextEditor (date/time/percent), and beforeChange already covers it:

```
beforeChange(changes) {
  changes.forEach((c) => { if (c[3] === '') c[3] = null; });
}
```

Closing as expected.
